stellarphot/photometry/profiles.py

```diff
@@ -0,0 +1,305 @@
+        self._cen = find_center(data, center_approx, cutout_size=cutout_size)
+        self._data = data
+        self._cutout = Cutout2D(data, self._cen, (cutout_size, cutout_size))
+        self._max_iters = max_iters
+        if profile_radius is None:
+            profile_radius = cutout_size // 2
+
+        radii = np.linspace(0, profile_radius, profile_radius + 1)
+        # Get a rough profile without background subtraction
+        self._radial_profile = RadialProfile(self._data, self._cen, radii)
+
+        self._sky_area = None
+
+        # Do proper background subtraction for the final radial profile
+        self._radial_profile = RadialProfile(
+            data - self.sky_pixel_value, self._cen, radii
+        )
+
+    @property
+    def center(self):
+        """
+        x, y position of the center of the star.
+        """
+        return self._cen
+
+    @property
+    def FWHM(self):
+        """
+        Full-width half-max of the radial profile.
+        """
+        return self.radial_profile.gaussian_fwhm
+
+    @property
+    def HWHM(self):
+        """
+        Half-width half-max of the radial profile.
+        """
+        return self.FWHM / 2
+
+    @property
+    def cutout(self):
+        """
+        Cutout image around the star.
+        """
+        return self._cutout
+
+    @property
+    def radial_profile(self):
+        """
+        Radial profile of the star.
+        """
+        return self._radial_profile
+
+    @property
+    def max_iters(self):
+        """
+        Maximum number of iterations to go through in finding the center.
+        """
+        return self._max_iters
+
+    @lazyproperty
+    def normalized_profile(self):
+        """
+        Radial profile scaled to have a maximum of 1.
+        """
+        # This seems to be what photutils does under the hood
+        return self.radial_profile.profile / self.radial_profile.profile.max()
+
+    @lazyproperty
+    def pixel_values_in_profile(self):
+        """
+        Pixel values in the radial profile.
+        """
+        radii = []
+        pixel_values = []
+        for rad, ap in zip(self.radial_profile.radius, self.radial_profile.apertures):
+            ap_mask = ap.to_mask(method="center")
+            ap_data = ap_mask.multiply(self._data)
+            good_data = ap_data != 0
+            pixel_values.extend(ap_data[good_data].flatten())
+            radii.extend([rad] * good_data.sum())
+        radii = np.array(radii)
+        pixel_values = np.array(pixel_values)
+        return radii, pixel_values
+
+    @lazyproperty
+    def curve_of_growth(self):
+        """
+        Curve of growth for the star.
+        """
+        self._cog = CurveOfGrowth(
+            self._data - self.sky_pixel_value,
+            self.center,
+            self.radial_profile.radii + 1,
+        )
+        return self._cog
+
+    @lazyproperty
+    def sky_pixel_value(self):
+        """
+        Pixel values for the sky, i.e. outside the star.
```

Comment:
OK, while likely to be outside the star, you might want to be clear you are assuming 3*FWHM is outside enough...